game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/battlegroup/FiringSquadron.java

```diff
@@ -0,0 +1,241 @@
+package games.strategy.triplea.delegate.battle.steps.fire.battlegroup;
+
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.triplea.java.PredicateBuilder;
+
+/** Holds predicates for finding units in a squadron and what they target */
```

Comment:
A 'spec' sounds like a config rather than an collection of units. I would suggest these two options:
1. Change the name of 'FiringGroup', do that in another PR and then you'd be free to use that name here. Perhaps we can name 'FiringGroup' to be a bit better, as the 'FiringGroup' is more of a configuration, or more of a computation that says these units can hit these other units, but it is not actually a group of units per say.
2. Document 'FiringSquadron', it is possible we can bake that into the terminology.

Overall this is a naming & vocabulary exercise, not easy, we all know code is much easier to understand when the vocabulary is good and descriptive.

Reply:
We can't change the name of `FiringGroup` because it is part of saves.  Otherwise, I would have renamed it.  And the existing `FiringGroup` is a group of units.